nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/script/ScriptingComponentHelper.java

```diff
@@ -265,7 +265,11 @@ public void setupVariables(final PropertyContext context) {
         scriptEngineName = context.getProperty(SCRIPT_ENGINE).getValue();
         scriptPath = context.getProperty(ScriptingComponentUtils.SCRIPT_FILE).evaluateAttributeExpressions().getValue();
         scriptBody = context.getProperty(ScriptingComponentUtils.SCRIPT_BODY).getValue();
-        modules = context.getProperty(ScriptingComponentUtils.MODULES).evaluateAttributeExpressions().asResources().flattenRecursively();
+        if (scriptEngineName.equals("python")) {
```

Comment:
Suggest you make this case-insensitive allowing some future-proofing. I see other engines start with a capital.

Reply:
committed fix that tries to incorporate both of your comments :)

Thank you very much @dan-s1 for your comments!

I agree that some sort of enum/ refactoring to incorporate the enum would be nice although this maybe pushing the scope of this ticket ( To be sure this is a valid area that could use improvement, but this has been around for a while pre-dating this ticket)